interface/client/lib/helpers/helperFunctions.js

```diff
@@ -50,6 +50,28 @@ Helpers.formatUrl = function(url){
 };
 
 /**
+Sanatizes URLs to prevent phishing and XSS attacks
+
+@method sanatizeUrl
+@param {String} url
+**/
+Helpers.sanatizeUrl = function(url){
+    var parser;
+
+    if(_.isString(url)) {
+        parser = document.createElement('a');
+        parser.href = url;
+    } else if(_.isObject(url) && url.protocol) {
+        parser = url;
+    }
+
+    parser.protocol = parser.protocol.replace(/^(javascript:)?(data:)?(:)?/i, 'http:');
```

Comment:
Here's an [example](http://martin.swende.se/tools/inject_it.js/#aW5pdEFwcCh3aW5kb3cpO2FkZENvZGVNaXJyb3IodW5lc2NhcGUoJ3NhbmF0aXplVXJsJTIwJTNEJTIwZnVuY3Rpb24lMjh1cmwlMjklN0IlMEElMjAlMjAlMjAlMjB2YXIlMjBwYXJzZXIlMjAlM0QlMjBkb2N1bWVudC5jcmVhdGVFbGVtZW50JTI4JTI3YSUyNyUyOSUzQiUwQSUyMCUyMCUyMCUyMHBhcnNlci5ocmVmJTIwJTNEJTIwJTIyJTIyK3VybCUzQiUyMC8vU3RyaW5naWZ5JTIwVVJMJTBBJTIwJTA5JTBBJTIwJTIwJTIwJTIwaWYlMjglMjBwYXJzZXIucHJvdG9jb2wlMjAlM0QlM0QlMjAlMjJqYXZhc2NyaXB0JTNBJTIyJTIwJTdDJTdDJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwcGFyc2VyLnByb3RvY29sJTIwJTNEJTNEJTIwJTIyZGF0YSUzQSUyMiUyOSUwQSUyMCUyMCUyMCUyMCU3QiUwQSUyMCUyMCUyMCUyMCUyMCUyMHBhcnNlci5wcm90b2NvbCUyMCUzRCUyMCUyMmh0dHBzJTNBLy8lMjIlMEElMjAlMjAlMjAlMjAlN0QlMEElMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBwYXJzZXIudG9TdHJpbmclMjglMjklM0IlMEElMjAlN0QlM0IlMEElMEFzaG93JTI4c2FuYXRpemVVcmwlMjglMjJqYXZhcyU1Q3RjcmlwVCUzQWFsZXJ0JTI4MSUyOSUyMiUyOSUyOScpKTsNCnVwZGF0ZU91dHB1dCgpOw0K) which is even more robust, although I'd still prefer whitelisting protos instead of blacklisting